src/JuMP.jl

```diff
@@ -278,13 +278,103 @@ Base.next(x::AbstractJuMPScalar, state) = (x, true)
 Base.done(::AbstractJuMPScalar, state) = state
 Base.isempty(::AbstractJuMPScalar) = false
 
+"""
+    AbstractShape
+
+Abstract vectorizable shape. Given a vectorized form of an object of shape
```

Comment:
"vectorized" is used in lots of contexts. Maybe "flat vector form" would be less ambiguous?